### ISSUE TITLE
aws.Service: allow deploying ECS tasks into public subnets in case VPC has no NATs

### ISSUE THREAD

**Issue by Dizzzmas** [sst/ion#901]
[PR](https://github.com/sst/ion/pull/902)
## Motivation

When placing `aws.Cluster` into a VPC with `nat: false` I'd like to be able to add `aws.Service` to the cluster with ECS tasks getting deployed into the public subnets.
That's useful for workloads that don't need access to private resources and allows to save on NAT fees.

e.g.
```typescript
    const vpc = new sst.aws.Vpc("TestClusterVpc", {
      nat: false,
    });
    const cluster = new sst.aws.Cluster("TestECSCluster", { vpc });
    cluster.addService("TestClusterService", {
      image: ".",
      public: {
        ports: [
          { listen: "80/http" },
        ]
      },
      transform: {
         service: {
          networkConfiguration: {
            subnets: vpc.publicSubnets,
            assignPublicIp: true
          }
        }
      }
    })
```

Currently this errors with:
```
The VPC configured for the service does not have NAT enabled. Enable NAT by configuring "nat" on the "sst.aws.Vpc" component.
```

## Proposed solution

When doing `cluster.addService` detect if the cluster's VPC has any NATs. If not - place the ECS tasks into a public subnet.
i.e. update the service configuration with the following
```
      networkConfiguration: {
        assignPublicIp: true,
        subnets: vpc.publicSubnets,
      }
```

**Comment by digitalcraft** [comment]
this has been sitting here for 2 weeks. can we get this merged?

**Comment by zammitjames** [comment]
cc @fwang

**Comment by jayair** [comment]
I think we are doing some work around containers in general starting next week. But I'll ping Frank.

**Comment by fwang** [comment]
Just released a new version of `Cluster` in [v3.1.16](https://github.com/sst/ion/releases/tag/v3.1.16). The new `Cluster` deploys services to the public subnets by default.
